app.rb

```diff
@@ -27,27 +44,33 @@ class OpenidConnectRelyingParty < Sinatra::Base
       nonce: SecureRandom.hex,
       prompt: 'select_account',
     }.to_query
-
-    erb :index, locals: { authorization_url: authorization_url }
   end
 
-  get '/auth/result' do
-    token_response = token(params[:code])
-    userinfo_response = userinfo(token_response[:id_token])
-
-    erb :success, locals: { userinfo: userinfo_response }
+  def openid_configuration
+    @openid_configuration ||= begin
+      response = openid_configuration_response
+      if response.code == 200
+        json(response.body)
+      else
+        false
+      end
+    end
   end
 
-  private
+  def openid_configuration_response
+    HTTParty.get(
+      URI.join(SERVICE_PROVIDER, '/.well-known/openid-configuration'),
+      basic_auth: BASIC_AUTH
+    )
+  end
 
-  def openid_configuration
-    @openid_configuration ||= begin
-      json(
-        HTTParty.get(
-          URI.join(SERVICE_PROVIDER, '/.well-known/openid-configuration'),
-          basic_auth: BASIC_AUTH
-        ).body
-      )
+  def openid_configuration_error
+    response = openid_configuration_response
+    if response.code == 401
+      "Error: #{SERVICE_PROVIDER} responded with #{response.code}.
+       Check basic authentication in environment variables.)"
+    else
+      "Error: #{SERVICE_PROVIDER} responded with #{response.code}."
```

Comment:
I don't think we have anything else, we could print the response but it may not be useful?